The "RealmBuildTasks.WeaveRealmAssembly" task failed unexpectedly

#### Goal

I'm trying to build my Xamarin.Forms project on Windows, using Visual Studio 2017 and targeting iOS, so building happens on a remote macOS installation.

#### Expected Results

App builds fine, just like it does in Xamarin Studio.

#### Actual Results

All iOS builds (debug/release/simulator/device) fail with the following, even after reinstalling Realm nuget package.

```
Realm.Database.targets(16,5): error MSB4018: The "RealmBuildTasks.WeaveRealmAssembly" task failed unexpectedly.
Realm.Database.targets(16,5): error MSB4018: System.Exception: Realm is referenced by your project but was not found in C:\Users\didan\Work\MyApp\iOS\bin\iPhoneSimulator\Debug\. Make sure that CopyLocal is set to True and try again.
Realm.Database.targets(16,5): error MSB4018:    at RealmBuildTasks.WeaveRealmAssembly.WeaveAssembly(String name, AssemblyDefinition currentAssembly)
Realm.Database.targets(16,5): error MSB4018:    at RealmBuildTasks.WeaveRealmAssembly.Execute()
Realm.Database.targets(16,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
Realm.Database.targets(16,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()
```

#### Steps & Code to Reproduce

In my case, the project was created in Xamarin Studio. It's a Xamarin.Forms project, targeting both iOS and Android.

Compiling in Xamarin Studio works fine, but as soon as I switch to Visual Studio 2017, the reference error is thrown.

Do note that both Realm and Realm.Database have the "Copy Local" option set to "true", like they should.

#### Version of Realm and tooling
Realm version(s): 1.1.0

Xamarin/Visual Studio version: v2.3.3.193 / Visual Studio 2017 (Release)

Which operating system version and device: Windows 10, macOS Sierra and iPhone/iPhone Simulator

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,10 @@
+x.y.z (TBD)
+------------------
+
+### Bug fixes
+
+- Resolved an issue that prevented compiling for iOS on Visual Studio. (#1277)
+
 1.1.0 (2017-03-03)
 ------------------
 

--- a/NuGet/Realm.Database/Realm.Database.targets
+++ b/NuGet/Realm.Database/Realm.Database.targets
@@ -9,6 +9,8 @@
     Condition="'$(TargetFrameworkIdentifier)' == 'Xamarin.iOS' AND '$(OutputType)' == 'Exe'">
     <ItemGroup>
       <FilteredReferences Include="@(ReferencePath)" Condition="('%(Filename)%(Extension)' == 'Realm.dll') Or ('%(Filename)%(Extension)' == 'Realm.Sync.dll')" />
+      <ReferencePath Include="$(OutputPath)Realm.dll" Condition="Exists('$(OutputPath)Realm.dll')"/>
+      <ReferencePath Include="$(OutputPath)Realm.Sync.dll" Condition="Exists('$(OutputPath)Realm.Sync.dll')"/>
       <ReferencePath Remove="@(FilteredReferences)" />
       <ReferenceCopyLocalPaths Remove="@(FilteredReferences)" />
     </ItemGroup>

--- a/NuGet/Realm.Database/Realm.Database.targets
+++ b/NuGet/Realm.Database/Realm.Database.targets
@@ -5,8 +5,7 @@
     AssemblyFile="$(MSBuildThisFileDirectory)../tools/Realm.BuildTasks.dll" />
   <Target
     Name="WeaveRealmAssemblyTarget" 
-    AfterTargets="AfterCompile"
-    BeforeTargets="DeployOutputFiles"
+    BeforeTargets="_CompileToNative"
     Condition="'$(TargetFrameworkIdentifier)' == 'Xamarin.iOS' AND '$(OutputType)' == 'Exe'">
     <ItemGroup>
       <FilteredReferences Include="@(ReferencePath)" Condition="('%(Filename)%(Extension)' == 'Realm.dll') Or ('%(Filename)%(Extension)' == 'Realm.Sync.dll')" />

--- a/Platform.XamarinIOS/Tests.XamarinIOS/Tests.XamarinIOS.csproj
+++ b/Platform.XamarinIOS/Tests.XamarinIOS/Tests.XamarinIOS.csproj
@@ -160,6 +160,8 @@
   <Target Name="WeaveRealmAssemblyTarget" BeforeTargets="_CompileToNative" >
     <ItemGroup>
       <FilteredReferences Include="@(ReferencePath)" Condition="('%(Filename)%(Extension)' == 'Realm.dll') Or ('%(Filename)%(Extension)' == 'Realm.Sync.dll')" />
+      <ReferencePath Include="$(OutputPath)Realm.dll" Condition="Exists('$(OutputPath)Realm.dll')"/>
+      <ReferencePath Include="$(OutputPath)Realm.Sync.dll" Condition="Exists('$(OutputPath)Realm.Sync.dll')"/>
       <ReferencePath Remove="@(FilteredReferences)" />
       <ReferenceCopyLocalPaths Remove="@(FilteredReferences)" />
     </ItemGroup>

--- a/Platform.XamarinIOS/Tests.XamarinIOS/Tests.XamarinIOS.csproj
+++ b/Platform.XamarinIOS/Tests.XamarinIOS/Tests.XamarinIOS.csproj
@@ -157,7 +157,7 @@
   <UsingTask 
 		TaskName="RealmBuildTasks.WeaveRealmAssembly" 
 		AssemblyFile="$(SolutionDir)Weaver\Realm.BuildTasks\bin\$(Configuration)\Realm.BuildTasks.dll" />
-  <Target Name="WeaveRealmAssemblyTarget" AfterTargets="AfterCompile" BeforeTargets="DeployOutputFiles" >
+  <Target Name="WeaveRealmAssemblyTarget" BeforeTargets="_CompileToNative" >
     <ItemGroup>
       <FilteredReferences Include="@(ReferencePath)" Condition="('%(Filename)%(Extension)' == 'Realm.dll') Or ('%(Filename)%(Extension)' == 'Realm.Sync.dll')" />
       <ReferencePath Remove="@(FilteredReferences)" />

--- a/Settings.StyleCop
+++ b/Settings.StyleCop
@@ -5,6 +5,7 @@
       <Value>awaitable</Value>
       <Value>changeset</Value>
       <Value>databinding</Value>
+      <Value>errored</Value>
       <Value>finalizer</Value>
       <Value>Fody</Value>
       <Value>hpp</Value>
@@ -15,6 +16,7 @@
       <Value>queryable</Value>
       <Value>runloop</Value>
       <Value>ulong</Value>
+      <Value>Uploadable</Value>
       <Value>var</Value>
       <Value>Xamarin</Value>
     </CollectionProperty>

--- a/Shared/Realm.Sync.Shared/Permissions/PermissionOfferResponse.cs
+++ b/Shared/Realm.Sync.Shared/Permissions/PermissionOfferResponse.cs
@@ -27,7 +27,7 @@ namespace Realms.Sync
     /// <remarks>
     /// Create an instance of <see cref="PermissionOfferResponse"/> using the provided <see cref="PermissionOffer.Token"/>
     /// and add it to the <see cref="User"/>'s ManagementRealm.
-    /// After that, you should subsribe to <see cref="RealmObject.PropertyChanged"/> to be notified when the server
+    /// After that, you should subscribe to <see cref="RealmObject.PropertyChanged"/> to be notified when the server
     /// processes the response.
     /// Once the request has been processed, the <see cref="Status"/>, <see cref="StatusMessage"/>, and
     /// <see cref="ErrorCode"/> will be updated accordingly.
@@ -80,7 +80,7 @@ namespace Realms.Sync
         public ErrorCode? ErrorCode => ErrorCodeHelper.GetErrorCode(StatusCode);
 
         /// <summary>
-        /// Gets the token thas was provided by the offering user.
+        /// Gets the token that was provided by the offering user.
         /// </summary>
         [Required]
         [MapTo("token")]

--- a/Shared/Realm.Sync.Shared/User.cs
+++ b/Shared/Realm.Sync.Shared/User.cs
@@ -100,7 +100,7 @@ namespace Realms.Sync
         /// <remarks>
         /// Users are persisted in a realm file within the application's sandbox.
         /// <para>
-        /// By default <see cref="User"/> objects are persisted and are additionaly protected with an encryption key stored
+        /// By default <see cref="User"/> objects are persisted and are additionally protected with an encryption key stored
         /// in the iOS Keychain when running on an iOS device (but not on a Simulator).
         /// On Android users are persisted in plaintext, because the AndroidKeyStore API is only supported on API level 18 and up.
         /// You might want to provide your own encryption key on Android or disable persistence for security reasons.

--- a/Weaver/Realm.BuildTasks/WeaveRealmAssembly.cs
+++ b/Weaver/Realm.BuildTasks/WeaveRealmAssembly.cs
@@ -158,7 +158,7 @@ namespace RealmBuildTasks
                 $"WeaveRealmAssembly: {message}",
                 string.Empty,
                 "BuildTasks",
-                MessageImportance.Normal));
+                MessageImportance.High));
         }
 
         private bool TryReadAssembly(string name, out AssemblyDefinition assembly, bool isRealmAssembly = true)

